includes/wp-admin/class-settings.php

```diff
@@ -226,25 +241,44 @@ public static function register_settings() {
 				'sanitize_callback' => array( Sanitize::class, 'url_list' ),
 			)
 		);
+
+		\register_setting(
+			'activitypub_welcome',
+			'activitypub_welcome_skip',
+			array(
+				'type'    => 'boolean',
+				'default' => false,
+			)
+		);
 	}
 
 	/**
 	 * Load settings page.
 	 */
 	public static function settings_page() {
+		$default_tab = \get_user_meta( \get_current_user_id(), 'activitypub_show_welcome_tab', true ) ? 'welcome' : 'settings';
 		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-		$tab = isset( $_GET['tab'] ) ? sanitize_key( $_GET['tab'] ) : 'welcome';
+		$tab = isset( $_GET['tab'] ) ? sanitize_key( $_GET['tab'] ) : $default_tab;
 
-		$settings_tabs = array(
-			'welcome'  => array(
+		// Redirect welcome tab to settings if skipped.
+		if ( 'welcome' === $tab && ! \get_user_meta( \get_current_user_id(), 'activitypub_show_welcome_tab', true ) ) {
+			$tab = 'settings';
+		}
+
+		$settings_tabs = array();
+
+		if ( \get_user_meta( \get_current_user_id(), 'activitypub_show_welcome_tab', true ) ) {
+			$settings_tabs['welcome'] = array(
 				'label'    => __( 'Welcome', 'activitypub' ),
 				'template' => ACTIVITYPUB_PLUGIN_DIR . 'templates/welcome.php',
-			),
-			'settings' => array(
-				'label'    => __( 'Settings', 'activitypub' ),
-				'template' => ACTIVITYPUB_PLUGIN_DIR . 'templates/settings.php',
-			),
+			);
+		}
+
+		$settings_tabs['settings'] = array(
+			'label'    => __( 'Settings', 'activitypub' ),
+			'template' => ACTIVITYPUB_PLUGIN_DIR . 'templates/settings.php',
 		);
```

Comment:
Nitpick, feel free to ignore: we could leave that in the initial `$settings_tabs` declaration

Reply:
but then the welcome page would be second or we would have to order it afterwards